### PR TITLE
Make `[p]streamalert list` show platform, allow to pass target channel to `[p]streamalert` commands

### LINKED ISSUE #3866: Make `[p]streamalert list` show platform, allow to pass target channel to `[p]streamalert` commands
# Feature request

#### Select the type of feature you are requesting:

- [ ] Cog
- [x] Command
- [ ] API functionality

#### Describe your requested feature

Cog: Streams

Command: `streamalert list`
Feature: Would be nice if this command also showed what platform the channel you add is on.

Command: `streamalert <platform> [channel_name] [channel]`
Feature: Maybe add so you can specify what channel you want the alert to be in so you don't have to 
use the command in the alert channel. Maybe same for the `channel` subcommand toggle.

### PR DESCRIPTION
**Pull request by Jackenmen**
### Description of the changes
Recreation of #4151 per [this comment](https://github.com/Cog-Creators/Red-DiscordBot/pull/4151#issuecomment-751285599). I addressed my review comments here as well, though my updates have not been tested yet.

---

Fixes #3866 

- Added argument to specify the discord channel where stream alerts appear
- Stream alert list shows the platform the channels were added on(twitch or youtube)

### CODE REVIEW

File: redbot/cogs/streams/streams.py [hunk 1]
```diff
@@ -448,7 +467,7 @@ async def stream_alert(self, ctx: commands.Context, _class, channel_name):
                     await ctx.send(_("That channel doesn't seem to exist."))
```
Comment (Kowlin) [inline redbot/cogs/streams/streams.py]:
```suggestion
                    await ctx.send(_("That stream doesn't seem to exist."))
```

**Review by Kowlin — DISMISSED** [review]
Besides the minor error text, it works just fine.

File: redbot/cogs/streams/streams.py (hunk 1 above)
Comment (Jackenmen) [reply redbot/cogs/streams/streams.py]: stream**er** perhaps?

Comment (Kowlin) [reply redbot/cogs/streams/streams.py]: Do what you like. ¯\\\_(ツ)\_/¯

**Review by Kowlin — APPROVED** [review]
Consistency! ~~(Inconsistent)~~

**Pull request merged**